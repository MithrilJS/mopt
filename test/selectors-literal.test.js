"use strict";

var code = require("./lib/code");

describe("Selectors", () => {
    describe("Literal selectors", () => {
        it("should support the empty selector", () =>
            expect(
                code(`m("")`)
            )
            .toMatchSnapshot()
        );

        it("should support tag selectors", () =>
            expect(
                code(`m("div")`)
            )
            .toMatchSnapshot()
        );
        
        it("should support class selectors", () =>
            expect(
                code(`m(".foo")`)
            )
            .toMatchSnapshot()
        );

        it("should support id selectors", () =>
            expect(
                code(`m("#foo")`)
            )
            .toMatchSnapshot()
        );
        
        it("should support empty attribute selectors", () =>
            expect(
                code(`m("div[fooga]")`)
            )
            .toMatchSnapshot()
        );

        it("should support attribute selectors", () =>
            expect(
                code(`m("[title=bar]")`)
            )
            .toMatchSnapshot()
        );

        it("should support single-quoted attribute selectors", () =>
            expect(
                code(`m("[title='bar']")`)
            )
            .toMatchSnapshot()
        );

        it("should support double-quoted attribute selectors", () =>
            expect(
                code(`m('[title="bar"]')`)
            )
            .toMatchSnapshot()
        );

        it("should ignore non-string selectors", () =>
            expect(
                code(`m(fooga)`)
            )
            .toMatchSnapshot()
        );
    });
});
