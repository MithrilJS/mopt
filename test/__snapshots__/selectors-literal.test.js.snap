// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selectors Literal selectors should ignore non-string selectors 1`] = `"m(fooga);"`;

exports[`Selectors Literal selectors should support attribute selectors 1`] = `"m.vnode(\\"div\\",undefined,{title:\\"bar\\"},[],undefined,undefined);"`;

exports[`Selectors Literal selectors should support class selectors 1`] = `"m.vnode(\\"div\\",undefined,{className:\\"foo\\"},[],undefined,undefined);"`;

exports[`Selectors Literal selectors should support double-quoted attribute selectors 1`] = `"m.vnode(\\"div\\",undefined,{title:\\"bar\\"},[],undefined,undefined);"`;

exports[`Selectors Literal selectors should support empty attribute selectors 1`] = `"m.vnode(\\"div\\",undefined,{fooga:true},[],undefined,undefined);"`;

exports[`Selectors Literal selectors should support id selectors 1`] = `"m.vnode(\\"div\\",undefined,{id:\\"foo\\"},[],undefined,undefined);"`;

exports[`Selectors Literal selectors should support single-quoted attribute selectors 1`] = `"m.vnode(\\"div\\",undefined,{title:\\"bar\\"},[],undefined,undefined);"`;

exports[`Selectors Literal selectors should support tag selectors 1`] = `"m.vnode(\\"div\\",undefined,undefined,[],undefined,undefined);"`;

exports[`Selectors Literal selectors should support the empty selector 1`] = `"m.vnode(\\"div\\",undefined,undefined,[],undefined,undefined);"`;
